features/net/network-socket/WiFiInterface.h

```diff
@@ -43,28 +42,47 @@ class WiFiInterface: public NetworkInterface
 public:
     /** Start the interface
      *
-     *  Attempts to connect to a WiFi network. If passphrase is invalid,
-     *  NSAPI_ERROR_AUTH_ERROR is returned.
+     *  Attempts to connect to a WiFi network.
      *
      *  @param ssid      Name of the network to connect to
      *  @param pass      Security passphrase to connect to the network
      *  @param security  Type of encryption for connection
-     *  @return          0 on success, negative error code on failure
+     *  @return          0 on success, or error code on failure
      */
-    virtual int connect(const char *ssid, const char *pass, nsapi_security_t security = NSAPI_SECURITY_NONE) = 0;
+    virtual nsapi_error_t connect(const char *ssid, const char *pass, nsapi_security_t security = NSAPI_SECURITY_NONE) = 0;
 
     /** Stop the interface
      *
-     *  @return          0 on success, negative error code on failure
+     *  @return          0 on success, or error code on failure
      */
-    virtual int disconnect() = 0;
+    virtual nsapi_error_t disconnect() = 0;
 
     /** Get the local MAC address
      *
      *  @return         Null-terminated representation of the local MAC address
      */
     virtual const char *get_mac_address() = 0;
-};
 
+    /** Get the current WiFi interface state
+     *
+     * @returns         Interface state enum
+     */
+    virtual nsapi_if_state_t get_state() = 0;
+
+    /** Gets the current radio signal strength for active connection
+     *
+     * @return Connection strength in dBm (negative value), or 0 if measurement impossible
+     */
+    virtual int16_t get_rssi() = 0;
+
+    /** Scan for available networks
+     *
+     * @param  ap    Pointer to allocated array to store discovered AP
+     * @param  count Size of allocated @a res array, or 0 to only count available AP
+     * @return       Number of entries in @a, or if @a count was 0 number of available networks, negative on error
+     *               see @a nsapi_error
+     */
+    virtual int scan(wifi_ap_t *res, unsigned count) = 0;
```

Comment:
This is not yet implemented to esp8266-driver, or is it ? I just look at quickly AT command which give you list of access points, and it depend how implement will be done that is it possible to get one by one those results easily or not. At least ESP8266 give you whole list of access points so you have to reserve memory for all of them dynamically before returning back.. but it probably would be much faster if you could get results one by one as soon as you get result from wifi chip.. there is cons and prons in both ways..